python/Makefile.mk

```diff
@@ -1,2 +1,17 @@
 include python/examples/Makefile.mk
 include python/ola/Makefile.mk
+
+python/PyCompileTest.sh: python/Makefile.mk
+	mkdir -p $(top_builddir)/python
+# 	restore this line when py3 compat is done for whole tree
+#	echo "$(PYTHON) -m compileall -f tools scripts python include data; exit \$$?" > $(top_builddir)/python/PyCompileTest.sh
+	echo "$(PYTHON) -m compileall -f python data; exit \$$?" > $(top_builddir)/python/PyCompileTest.sh
```

Comment:
I acknowledge the point, but it also seems a bit odd to have that cruft added to include/Makefile.mk just to compile that one script.  The various coverage/flake8 etc tests generally run once from the root dir; in theory it could be put in Makefile.am, but that seems odd.  IMHO this was the best compromise, but if you feel strongly about it, I could put it in the various subdirs that need it.  (after I check the fix for the race condition that's causing the mac build to fail right now)

Reply:
@peternewman I haven't changed this code.  Thoughts?

Reply:
Sorry, missed this. I'd probably prefer to have it per folder in the various Makefile.mk's like our other code. We could have it separately in the Makefile like the lint stuff, but this feels a bit more important than that so should probably run against every test run.

Or put it in Makefile.am so it's one test run at the top level.

Will this need the imports to be present to work? In which case it should probably go in the subfolders so the logic for e.g. numpy for the RDM tests is kept in one place? If not putting it in the root and compiling everything seems like a win.

We could possibly even have some logic to compile **everything** if it's running a Python 2 build, but that might be a lot of faff.

Reply:
I tried to split the difference of leaving the script building in python/Makefile.mk but having a variable that assembled directories that should be compiled and are set in the particular directories that need it.  That seems to me like it captures the spirit of having the opt-in be local while maintaining a single copy of the script and not adding test-generation stuff into the root Makefile.am.

Reply:
Okay, fair enough if you think we should keep Makefile.am cleaner.